docs/userguide/serialization.md

```diff
@@ -0,0 +1,385 @@
+
+assert t1 == t2 == t3
+```
+
+### Tagged subclasses
+Due to the leading design philosophy behind BayBE to provide its users easy access
+to a broad range of tools, you typically have the choice between several modelling 
+alternatives when building your objects.
+For example, when describing the degrees of freedom of your experimental campaign,
+you can chose from several different [parameter types](/userguide/parameters).
+
+While this is offers great flexibility, it comes with a challenge for deserialization
+because you cannot know a priori which concrete object subclass is contained 
+in an incoming serialization string on the receiving end.
+Instead, you oftentimes need to be able to process the incoming string dynamically.
+
+For example, consider the following string, which perfectly mirrors the signatures of 
+both
+{meth}`CategoricalParameter <baybe.parameters.categorical.CategoricalParameter.__init__>` and 
+{meth}`TaskParameter <baybe.parameters.categorical.TaskParameter.__init__>`:
+```python
+parameter_str = """
+{
+    "name": "setting",
+    "values": ["low", "high"]
+}
+"""
+```
+Unless you are aware of the specific purpose for which the string was created,
+calling one of the classes' constructors directly is impossible because you 
+simply do not know which one to chose.
+A similar situation arises with [nested objects](#NESTED_OBJECTS) because resorting to
+an explicit constructor call of a hand-selected subclass is only possible at the
+highest level of the hierarchy, whereas the inner object types would remain unspecified.
+
+The problem can be easily circumvented using an explicit subclass resolution 
+mechanism, i.e., by tagging the respective subclass in an additional `type` field that
+holds the class' name.
+This allows to deserialize the object from the corresponding base class instead,
+mirroring the flexibility of specifying subtypes to your configuration file:
+
+```python
+from baybe.parameters.base import Parameter
+from baybe.parameters import CategoricalParameter, TaskParameter
+
+categorial_parameter = CategoricalParameter(name="setting", values=["low", "high"])
+categorical_parameter_str = """
+{
+    "type": "CategoricalParameter",
+    "name": "setting",
+    "values": ["low", "high"]
+}
+"""
+categorical_parameter_reconstructed = Parameter.from_json(categorical_parameter_str)
+assert categorial_parameter == categorical_parameter_reconstructed
+
+task_parameter = TaskParameter(name="setting", values=["low", "high"])
+task_parameter_str = """
+{
+    "type": "TaskParameter",
+    "name": "setting",
+    "values": ["low", "high"]
+}
+"""
+task_parameter_reconstructed = Parameter.from_json(task_parameter_str)
+assert task_parameter == task_parameter_reconstructed
+```
+
+```{note} 
+When serializing an object that belongs to a class hierarchy, BayBE automatically
+injects the `type` field into the serialization string to enable frictionless deserialization
+at a later stage.
+```
+
+### Using abbreviations
```

Comment:
@Scienfitz, we made the `_abbreviation` field private in #191 but now I notice: how should people even figure out if they are allowed to use an abbreviation? As you can see, I wrote ``Classes that have an `_abbreviation` class`` but unless digging into the code, people wouldn't know because the field doesn't even show up in the doc (since private). Should we make it public? Or any better ideas how we can inform people?

Reply:
makes sense to make it public so that stuff at least is `discoverable`

Reply:
fixed